pkg/services/store/object/tests/server_integration_test.go

```diff
@@ -38,24 +33,37 @@ type objectVersionMatcher struct {
 }
 
 func timestampInRange(ts int64, tsRange []time.Time) bool {
-	return ts >= tsRange[0].Unix() && ts <= tsRange[1].Unix()
+	low := tsRange[0].UnixMilli() - 1
+	high := tsRange[1].UnixMilli() + 1
+	return ts >= low && ts <= high
 }
 
 func requireObjectMatch(t *testing.T, obj *object.RawObject, m rawObjectMatcher) {
 	t.Helper()
 	require.NotNil(t, obj)
 
 	mismatches := ""
-	if m.grn != nil && !obj.GRN.Equals(m.grn) {
-		mismatches += fmt.Sprintf("expected: %v, actual: %v\n", m.grn, obj.GRN)
+	if m.grn != nil {
+		if m.grn.TenantId > 0 && m.grn.TenantId != obj.GRN.TenantId {
+			mismatches += fmt.Sprintf("expected tenant: %d, actual: %d\n", m.grn.TenantId, obj.GRN.TenantId)
```

Comment:
this seems like a lot of logic for a test helper. were there issues with using `require`?